components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -421,17 +421,13 @@ private void addUserAttributesToCache(OAuth2AccessTokenReqDTO tokenReqDTO, OAuth
             AuthorizationGrantCacheKey newCacheKey = new AuthorizationGrantCacheKey(tokenRespDTO.getAccessToken());
             if (authorizationGrantCacheEntry != null) {
                 authorizationGrantCacheEntry.setTokenId(tokenRespDTO.getTokenId());
-                if (AuthorizationGrantCache.getInstance().getValueFromCacheByToken(newCacheKey) == null) {
-                    if (log.isDebugEnabled()
-                            && IdentityUtil.isTokenLoggable(IdentityConstants.IdentityTokens.ACCESS_TOKEN)) {
-                        log.debug("No AuthorizationGrantCache entry found for the access token(hashed):" +
-                                DigestUtils.sha256Hex(newCacheKey.getUserAttributesId()) + ", hence adding to cache");
-                    }
-                    AuthorizationGrantCache.getInstance().addToCacheByToken(newCacheKey, authorizationGrantCacheEntry);
-                    AuthorizationGrantCache.getInstance().clearCacheEntryByCode(oldCacheKey);
-                } else {
-                    //if the user attributes are already saved for access token, no need to add again.
+                if (log.isDebugEnabled()
```

Comment:
if (log.isDebugEnabled()){
 +                    if(IdentityUtil.isTokenLoggable(IdentityConstants.IdentityTokens.ACCESS_TOKEN)) {
 +                    log.debug("Adding AuthorizationGrantCache entry for the access token(hashed):" +
 +                            DigestUtils.sha256Hex(newCacheKey.getUserAttributesId()));
}else {
log.debug("Adding AuthorizationGrantCache entry for the access token(hashed):"
}
                  }